admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -83,18 +83,33 @@ export class AnalyticsService {
     resolution: Resolution,
   ): MetricEntity[] {
     const timestampKeys = Object.keys(timestampMap);
-    let currentTime = dayjs.unix(Number(timestampKeys.at(0))).toDate();
-    const endTime = dayjs
+    let current = dayjs.unix(Number(timestampKeys.at(0))).toDate();
+    const end = dayjs
       .unix(Number(timestampKeys.at(timestampKeys.length - 1)))
       .toDate();
 
-    while (currentTime <= endTime) {
-      const key = Math.floor(currentTime.getTime() / 1000);
+    let manipulate: ManipulateType;
+    switch (resolution) {
+      case Resolution.Hour:
+        manipulate = Resolution.Hour;
```

Comment:
Shouldn't this be `ManipulateType.Hour`? (and same in the other cases below)

Reply:
ManipulateType is a type and not an enum unfortunately. Dayjs does not provide enums for their types as far as I know.